### DISCUSSION TITLE
Please make sure your database server is running at `127.0.0.1`:`5432`.

### DISCUSSION THREAD

**Discussion by lonewolf9277** [prisma/prisma#22671] [Q&A]
### Bug description

I have a discord bot deployed on digital ocean. The issue is that it all runs fine with postgres but after some time prisma stops connecting to the database. When this happens I restart postgres and prisma starts communicating with the database again.
![image](https://github.com/prisma/prisma/assets/3754311/0a1943b7-5f18-4d55-8ec1-86e5ff33db84)

### Prisma information

```prisma
// Add your schema.prisma
```

```ts
// Add your code using Prisma Client
```

### Environment & setup

- OS: Ubuntu 20.04 LTS
- Database: [PostgreSQL]
- Node.js version: [v18.19.0]

### Prisma Version

```
5.8.1
```

**Comment by ludralph** [comment]
Hi @lonewolf9277 👋 

Thank you for raising this question.

Can you please increase the timeout by adding `?connect_timeout=300` to the databse url? This was recommended in this [open issue](https://github.com/prisma/prisma/issues/8223) for this same problem.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by lonewolf9277** [reply]
I tried this yesterday but today it happened again.

**Reply by ludralph** [reply]
Hi @lonewolf9277, I can see that the URL points to a localhost (127.0.0.1:5432). Are you connecting to a local database from within Digital Ocean?

**Reply by lonewolf9277** [reply]
Yes postgresql is installed in the droplet. Is that the issue?

**Reply by ludralph** [reply]
Hi @lonewolf9277 

You're experiencing an intermittent connection issue between Prisma and your PostgreSQL database on DigitalOcean. One common cause of this issue is connection pool exhaustion. Prisma might be opening connections but not properly closing them, leading to a situation where no new connections can be established. To mitigate this, you can try the following:

- Add connection pool settings to your DATABASE_URL:
```
postgresql://USER:PASSWORD@HOST:PORT/DATABASE?pgbouncer=true&connection_limit=10&pool_timeout=20
```

This sets a connection limit and a pool timeout, which can help manage connections more effectively. 

- Consider using PgBouncer, a connection pooler for PostgreSQL. DigitalOcean offers PgBouncer as part of their managed database services. If you're not already using it, enabling PgBouncer can help manage connections more efficiently. Make sure to add `?pgbouncer=true` to your connection string if you use PgBouncer.

**Comment by lonewolf9277** [comment]
Hello any solutions to this?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;
